sdk/storage/azure-storage-blob/azure/storage/blob/_blob_client.py

```diff
@@ -159,6 +160,8 @@ def __init__(
             blob_name: str,
             snapshot: Optional[Union[str, Dict[str, Any]]] = None,
             credential: Optional[Union[str, Dict[str, str], "AzureNamedKeyCredential", "AzureSasCredential", "TokenCredential"]] = None,  # pylint: disable=line-too-long
+            *,
+            audience: Optional[BlobAudience] = None,
```

Comment:
I envision if we wanted to pass strictly the Audience's value (option 2) then somewhere in our init code here we would do:
```
if audience:
    kwargs['audience'] = str(audience)
```

So that we can extract the `.value` and slap this onto kwargs to get a free ride into `base_client` and consequently passed along to `Storage(+/-)BearerTokenCredential`